core/services/registrysyncer/orm.go

```diff
@@ -0,0 +1,286 @@
+				CapabilityId: to32Byte(capabilityId),
+				Config:       config,
+			}
+		}
+		t.IDsToDONs[DonID(id)] = kcr.CapabilitiesRegistryDONInfo{
+			Id:                       v.Id,
+			ConfigCount:              v.ConfigCount,
+			F:                        v.F,
+			IsPublic:                 v.IsPublic,
+			AcceptsWorkflows:         v.AcceptsWorkflows,
+			NodeP2PIds:               nodeP2PIds,
+			CapabilityConfigurations: configs,
+		}
+	}
+
+	t.IDsToNodes = make(map[p2ptypes.PeerID]kcr.CapabilitiesRegistryNodeInfo)
+	for k, v := range temp.IDsToNodes {
+		key, err := hex.DecodeString(k)
+		if err != nil {
+			return fmt.Errorf("failed to decode node key: %w", err)
+		}
+		var peerID p2ptypes.PeerID
+		copy(peerID[:], key[:32])
+
+		hashedCapabilityIds := make([][32]byte, len(v.HashedCapabilityIds))
+		for i, id := range v.HashedCapabilityIds {
+			b, err2 := hex.DecodeString(id)
+			if err2 != nil {
+				return fmt.Errorf("failed to decode hashedCapabilityId: %w", err2)
+			}
+			copy(hashedCapabilityIds[i][:], b[:32])
+		}
+
+		capabilitiesDONIds := make([]*big.Int, len(v.CapabilitiesDONIds))
+		for i, id := range v.CapabilitiesDONIds {
+			bigInt := new(big.Int)
+			bigInt.SetString(id, 10)
+			capabilitiesDONIds[i] = bigInt
+		}
+		signer, err := hex.DecodeString(v.Signer)
+		if err != nil {
+			return fmt.Errorf("failed to decode signer: %w", err)
+		}
+		p2pId, err := hex.DecodeString(v.P2pId)
+		if err != nil {
+			return fmt.Errorf("failed to decode p2pId: %w", err)
+		}
+		t.IDsToNodes[peerID] = kcr.CapabilitiesRegistryNodeInfo{
+			NodeOperatorId:      v.NodeOperatorId,
+			ConfigCount:         v.ConfigCount,
+			WorkflowDONId:       v.WorkflowDONId,
+			Signer:              to32Byte(signer),
+			P2pId:               to32Byte(p2pId),
+			HashedCapabilityIds: hashedCapabilityIds,
+			CapabilitiesDONIds:  capabilitiesDONIds,
+		}
+	}
+
+	t.IDsToCapabilities = make(map[HashedCapabilityID]kcr.CapabilitiesRegistryCapabilityInfo)
+	for k, v := range temp.IDsToCapabilities {
+		id, err := hex.DecodeString(k)
+		if err != nil {
+			return fmt.Errorf("failed to decode capability ID: %w", err)
+		}
+		var hashedId HashedCapabilityID
+		copy(hashedId[:], id[:32])
+
+		t.IDsToCapabilities[hashedId] = kcr.CapabilitiesRegistryCapabilityInfo{
+			HashedId:              hashedId,
+			LabelledName:          v.LabelledName,
+			Version:               v.Version,
+			CapabilityType:        v.CapabilityType,
+			ResponseType:          v.ResponseType,
+			ConfigurationContract: common.HexToAddress(v.ConfigurationContract),
+			IsDeprecated:          v.IsDeprecated,
+		}
+	}
+
+	return nil
+}
+
+type syncerORM struct {
+	ds   sqlutil.DataSource
+	lggr logger.Logger
+}
+
+func newORM(ds sqlutil.DataSource, lggr logger.Logger) syncerORM {
+	namedLogger := lggr.Named("RegistrySyncerORM")
+	return syncerORM{
+		ds:   ds,
+		lggr: namedLogger,
+	}
+}
+
+func (orm syncerORM) addState(ctx context.Context, state State) error {
```

Comment:
@vyzaldysanchez Can we add tests for this? I don't see anything that actually exercises the database updates

Reply:
The database operations are already being triggered on `TestSyncer_DBIntegration`, but I can add tests specific for the ORM.